<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />


<title>Transcriptomic analysis of CTC at different timepoints</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CIRCADIAN-CTC</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Data generation</li>
    <li>
      <a href="0_differential_expression_gsea_gsva.html">Run differential gene expression analysis, GSEA and GSVA</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Main results</li>
    <li>
      <a href="patients_ctc_counts_distribution.html">CTC counts distribution in breast cancer patients</a>
    </li>
    <li>
      <a href="br16_pca.html">PCA analysis of CTC gene expression in NSG-CDX-BR16 mice</a>
    </li>
    <li>
      <a href="br16_dge.html">Differential gene expression report for NSG-CDX-BR16 data</a>
    </li>
    <li>
      <a href="gsea_across_models.html">Report of GSEA and GSVA analysis in NSG-CDX-BR16, NSG-LM2, Patient and NSG-LM2 time kinetics</a>
    </li>
    <li>
      <a href="core_gene_sets.html">Analysis of genes from candidate pathways</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/TheAcetoLab/diamantopoulou-ctc-dynamics">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://www.theacetolab.com">
    <span class="fa fa-flask"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Transcriptomic analysis of CTC at different
timepoints</h1>
<h4 class="author">Francesc Castro-Giner</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>diamantopoulou-ctc-dynamics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabdiamantopoulouctcdynamicstreeaf1cf493b8202486e7ef9f4e4a37c4c21fded62etargetblankaf1cf49a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/diamantopoulou-ctc-dynamics/tree/af1cf493b8202486e7ef9f4e4a37c4c21fded62e" target="_blank">af1cf49</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabdiamantopoulouctcdynamicstreeaf1cf493b8202486e7ef9f4e4a37c4c21fded62etargetblankaf1cf49a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/diamantopoulou-ctc-dynamics/tree/af1cf493b8202486e7ef9f4e4a37c4c21fded62e" target="_blank">af1cf49</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/0_differential_expression_gsea_gsva.md
    Untracked:  analysis/about.md
    Untracked:  analysis/br16_dge.md
    Untracked:  analysis/br16_pca.md
    Untracked:  analysis/core_gene_sets.md
    Untracked:  analysis/gsea_across_models.md
    Untracked:  analysis/index.md
    Untracked:  analysis/license.md
    Untracked:  analysis/patients_ctc_counts_distribution.md
    Untracked:  data/differential_expression/
    Untracked:  data/patients/
    Untracked:  data/resources/
    Untracked:  data/sce/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/diamantopoulou-ctc-dynamics/8acaa646eebff98e6160ded07ba48258d757e6d7/docs/index.html" target="_blank">8acaa64</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/diamantopoulou-ctc-dynamics/blob/545ee28878a043453ec828b26ef18d6c110d8906/analysis/index.Rmd" target="_blank">545ee28</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-26
</td>
<td>
release v1.0
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/diamantopoulou-ctc-dynamics/545ee28878a043453ec828b26ef18d6c110d8906/docs/index.html" target="_blank">545ee28</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-26
</td>
<td>
release v1.0
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/diamantopoulou-ctc-dynamics/74b189131a39d49fd9e79eaee6c8e76575069de6/docs/index.html" target="_blank">74b1891</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/diamantopoulou-ctc-dynamics/blob/34ee51371a3786d0f18c3d2c11ec3e41b9f8ea4c/analysis/index.Rmd" target="_blank">34ee513</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-26
</td>
<td>
Update index with navbar and DOIs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/diamantopoulou-ctc-dynamics/bfb622b5aac934265e71dc66dd5a0985a4fca3d1/docs/index.html" target="_blank">bfb622b</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/diamantopoulou-ctc-dynamics/5dc02788c3fa12e0edce46d9a9eb670fec8cc19f/docs/index.html" target="_blank">5dc0278</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/diamantopoulou-ctc-dynamics/blob/c889911ccbd205a5c5ae3c0a0c13e4039f0b03b0/analysis/index.Rmd" target="_blank">c889911</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-26
</td>
<td>
Added final code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/diamantopoulou-ctc-dynamics/c889911ccbd205a5c5ae3c0a0c13e4039f0b03b0/docs/index.html" target="_blank">c889911</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-26
</td>
<td>
Added final code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/diamantopoulou-ctc-dynamics/a1ff2cdd3eeea7b524377b7838d06d6025710b7a/docs/index.html" target="_blank">a1ff2cd</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/diamantopoulou-ctc-dynamics/4277859c29b16dbb3e8abaa66d86e268df0a820e/docs/index.html" target="_blank">4277859</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/diamantopoulou-ctc-dynamics/blob/d9e45e9aab7c5c0ec86bc50c9a7f493aeab82662/analysis/index.Rmd" target="_blank">d9e45e9</a>
</td>
<td>
fcg-bio
</td>
<td>
2022-04-25
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="publication" class="section level2">
<h2>Publication</h2>
<p>Diamantopoulou, Z., Castro-Giner, F., Schwab F.D., Foerster C.,
Saini, M., Budinjas, S., Strittmatter, K., Krol, I., Seifert, B.,
Heinzelmann-Schwarz, V., Kurzeder, C., Rochlitz, C., Vetter, M., Weber,
W.P., &amp; Aceto N. The metastatic spread of breast cancer is
accomplished during sleep time. (2022)</p>
</div>
<div id="abstract" class="section level2">
<h2>Abstract</h2>
<p>We isolated live circulating tumor cells (CTCs) from a breast cancer
patient (number of CTCs = 36) and two breast cancer xenografts,
NSG-CDX-BR16 (number of CTCs = 138) and NSG-LM2 (number of CTCs = 108)
at different time points. Isolated CTCs were categorized into three
different types : single CTC, CTC clusters and CTC-white blood cells
(CTC-WBC) clusters. Amplified cDNA was prepared according to the
Smart-seq2 protocol. Libraries were prepared using with Nextera XT
(Illumina) and sequenced on Illumina NextSeq500 instrument in 75-bp
single-read mode.</p>
</div>
<div id="data-pre-processing" class="section level2">
<h2>Data pre-processing</h2>
<p>Raw data is available at Gene Expression Omnibus (GEO, NCBI;
accession number <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE180097">GSE180097</a>).
Data processing is computationally expensive and is not covered in this
repository. We provide description of the data pre-processing workflow
together with software version in the original publication. Processed
data, large result files, additional functions, references and metadata
are were archived at <a
href="https://doi.org/10.5281/zenodo.6358987"><img
src="https://zenodo.org/badge/DOI/10.5281/zenodo.6358987.svg"
alt="DOI" /></a></p>
</div>
<div id="data-and-code-availability" class="section level2">
<h2>Data and code availability</h2>
<p>To reproduce our analysis, first clone source code from the <a
href="https://github.com/TheAcetoLab/diamantopoulou-ctc-dynamics">GitHub
repository</a>. This repository is also archived at <a
href="https://doi.org/10.5281/zenodo.6484917"><img
src="https://zenodo.org/badge/DOI/10.5281/zenodo.6484917.svg"
alt="DOI" /></a></p>
<pre><code>git clone https://github.com/TheAcetoLab/diamantopoulou-ctc-dynamics.git</code></pre>
<p>Next, download processed data deposited in <a
href="https://doi.org/10.5281/zenodo.6358987">Zenodo</a> into the cloned
project folder ./data directory and untar the files.</p>
<pre><code>for file in *.tar.gz; do tar xzvf &quot;${file}&quot; &amp;&amp; rm &quot;${file}&quot;; done</code></pre>
</div>
<div id="reproducibility" class="section level2">
<h2>Reproducibility</h2>
<p>The results form our analyses are listed below in webpage format.
They were generated from R Markdown documents deposited in the <a
href="https://github.com/TheAcetoLab/diamantopoulou-ctc-dynamics">GitHub
repository</a>. The workflow of the analysis was created using the <a
href="https://cran.r-project.org/web/packages/workflowr/index.html">workflowr</a>
R package and can be reproduced in its totality using <a
href="https://cran.r-project.org/web/packages/workflowr/index.html">workflowr</a>
<a
href="https://jdblischak.github.io/workflowrBeta/reference/wflow_build.html">wflow_build</a>
command after the installation of the proper R packages. Session info,
including R and package versions, was automatically included at the end
of each analysis file.</p>
<p>Files containing pre-computed results from differential expression or
gene-set enrichment analyses were deposited in <a
href="https://doi.org/10.5281/zenodo.6358987"><img
src="https://zenodo.org/badge/DOI/10.5281/zenodo.6358987.svg"
alt="DOI" /></a>. In order to generate those files again change the
option <code>eval = FALSE</code> to <code>eval = TRUE</code> in the
specific code chunk from the R Markdown file.</p>
</div>
<div id="analyses" class="section level2">
<h2>Analyses</h2>
<ul>
<li><a href="0_differential_expression_gsea_gsva.html">Run differential
gene expression analysis, GSEA and GSVA</a></li>
<li><a href="patients_ctc_counts_distribution.html">CTC counts
distribution in breast cancer patients</a></li>
<li><a href="br16_pca.html">PCA analysis of CTC gene expression in
NSG-CDX-BR16 mice</a></li>
<li><a href="br16_dge.html">Differential gene expression report for
NSG-CDX-BR16 data</a></li>
<li><a href="gsea_across_models.html">Report of GSEA and GSVA analysis
in NSG-CDX-BR16, NSG-LM2, Patient and NSG-LM2 time kinetics</a></li>
<li><a href="core_gene_sets.html">Analysis of genes from candidate
pathways</a></li>
</ul>
</div>
<div id="paper-figures-and-tables" class="section level2">
<h2>Paper figures and tables</h2>
<ul>
<li>Main figures
<ul>
<li><a
href="patients_ctc_counts_distribution.html#Plot_polar_distribution">Figure
1a</a></li>
<li><a href="br16_pca.html#Biplot_PC4_and_PC5">Figure 3b</a></li>
<li><a
href="br16_dge.html#Heatmap_diferential_expression_NSG-CDX-BR16_mice">Figure
3c</a></li>
<li><a
href="gsea_across_models.html#Similarity_matrix_without_row_names">Figure
3d</a></li>
<li><a
href="gsea_across_models.html#GSEA_for_NSG-CDX-BR16_and_NSG-LM2">Figure
3e</a></li>
<li><a href="gsea_across_models.html#GSEA_for_Patient">Figure
3f</a></li>
<li><a
href="gsea_across_models.html#Average_GSVA_score_across_time_series">Figure
3g</a></li>
</ul></li>
<li>Extended data figures
<ul>
<li><a
href="br16_pca.html#Correlation_of_PC_eigenvectors_with_metadata">Extended
data Figure 6a</a></li>
<li><a
href="br16_dge.html#Correlation_DEG_single_CTC_versus_CTC_clusters_and_CTC-WBC">Extended
data Figure 6b</a></li>
<li><a
href="br16_dge.html#Compare_DEG_CTC_versus_CTC_clusters_and_CTC-WBC">Extended
data Figure 6c</a></li>
<li><a
href="gsea_across_models.html#Similarity_matrix_with_row_names">Extended
data Figure 6d</a></li>
<li><a
href="gsea_across_models.html#GSEA_for_NSG-CDX-BR16,_NSG-LM2_and_patient">Extended
data Figure 6e</a></li>
<li><a
href="gsea_across_models.html#GSVA_score_across_time_series">Extended
data Figure 6f</a></li>
<li><a
href="core_gene_sets.html#Expression_distribution_of_core_circadian_genes_in_NSG-CDX-BR16">Extended
data Figure 8a</a></li>
<li><a
href="core_gene_sets.html#Expression_of_TEAD_genes_in_CTCs_from_NSG-CDX-BR16">Extended
data Figure 8b</a></li>
<li><a
href="core_gene_sets.html#Expression_of_receptors_activated_by_circadian_rhythm_regulated_ligands">Extended
data Figure 9b</a></li>
</ul></li>
<li>Supplementary tables
<ul>
<li><a
href="br16_dge.html#Genes_differentially_expressed_in_CTCs_of_NSG-CDX-BR16">Supplementary
Table 2</a></li>
<li><a
href="gsea_across_models.html#NSG-CDX-BR16_GSEA_results">Supplementary
Table 3</a></li>
<li><a href="gsea_across_models.html#Table_GSEA_in_NSG-LM2">Source data
for Figure 3e, right panel</a></li>
<li><a href="gsea_across_models.html#Table_GSEA_in_Patients">Source data
for Figure 3f</a></li>
<li><a href="gsea_across_models.html#GSVA_table_of_results">Source data
for Figure 3g</a></li>
</ul></li>
</ul>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
